iocore/net/SSLConfig.cc

```diff
@@ -713,12 +727,22 @@ SSLConfigParams::getCTX(const char *client_cert, const char *key_file, const cha
   }
 
   ink_mutex_acquire(&ctxMapLock);
-  iter = ctx_map.find(key);
-  if (iter != ctx_map.end()) {
+  top_iter = top_level_ctx_map.find(top_level_key);
+  if (top_iter != top_level_ctx_map.end()) {
+    if (top_iter->second == nullptr) {
+      top_iter->second = new CTX_MAP;
```

Comment:
Is there any case come to this line? It looks like `top_iter->second` is created by above line 679, because `top_level_key` is not changed since then.

Reply:
The map is protected by the lock so I thought it makes sense to do another find on the map after releasing and grabbing the lock again.

Reply:
I missed the lock is released after the first lookup. Looking twice looks reasonable.